### ISSUE TITLE
Incorrect line numbers reported for .po files when msgid spans multiple lines

### ISSUE THREAD

**Issue by BHUVANSH855** [sphinx-contrib/sphinx-lint#168]
## Summary

Sphinx Lint reports incorrect line numbers for `.po` files. The reported
line does not point to the actual `msgstr` content line containing the
problem. Instead, it points near the beginning of the PO entry.

The offset is not constant—it increases as the corresponding `msgid`
occupies more physical lines (for example, when it is wrapped across
multiple lines).

This was mentioned as a follow-up item in #120:

> The only outstanding action item I see from this thread are line number
> annotations, but that should be handled in a separate issue IMO as its
> scope are PO files in general, not only the literal-blocks.

This issue is intended to track that behavior separately.

---

## Minimal reproduction

Create a file named `minimal2.po`:

```po
msgid ""
msgstr ""
"Content-Type: text/plain; charset=UTF-8\n"

#: ../../library/example.rst:10
msgid "Short one."
msgstr "Also short, and it has an ``unbalanced literal here."

#: ../../library/example.rst:22
msgid ""
"This original English sentence is deliberately long enough that xgettext "
"wraps it across two physical lines in the source file."
msgstr ""
"Line one of the translation, totally fine.\n"
"Line two, also fine, nothing wrong here.\n"
"Line three has the bug: an ``unbalanced literal here.\n"
"Line four is fine again, closing things off.\n"
```

Run:

```bash
sphinx-lint minimal2.po
```

Output:

```text
minimal2.po:5: found an unbalanced inline literal markup. (unbalanced-inline-literals-delimiters)
minimal2.po:11: found an unbalanced inline literal markup. (unbalanced-inline-literals-delimiters)
```

---

## Expected behavior

The reported line numbers should point to the actual translation lines that contain the errors.

For the example above:

| Entry | Actual line | Reported line |
|------|------------:|--------------:|
| Single-line `msgid` | 7 | 5 |
| Multi-line `msgid` | 16 | 11 |

The second entry demonstrates that the offset increases as the `msgid` spans additional physical lines.

---

## Initial investigation

The current implementation appears to map diagnostics relative to the start of each PO entry rather than the physical `msgstr` content lines.

See the investigation comment below for additional implementation details.

---

## Environment

- sphinx-lint version: 
- Python version: 

---

I'd be happy to work on a fix once the general approach is agreed upon.

**Comment by BHUVANSH855** [comment]
I did a bit more investigation into the current implementation.

`po2rst()` currently pads output using `entry.linenum`:

```python
while len(output) + 1 < entry.linenum:
    output.append("\n")
```

Using a minimal reproducer, I confirmed that `polib` reports:

| Entry | `entry.linenum` | Actual `msgstr` content starts |
|------|----------------:|-------------------------------:|
| 1 | 5 | 7 |
| 2 | 9 | 14 |

As a result, the generated RST maps the translation text to lines 5 and 9 instead of the actual translation lines, which explains the incorrect diagnostics.

I also investigated how `polib` populates `POEntry.linenum`. It is assigned when a new PO entry is created:

```python
self.current_entry = POEntry(linenum=self.current_line)
```

Later, when `msgstr` is parsed, only the translated text is stored:

```python
def handle_ms(self):
    self.current_entry.msgstr = unescape(self.current_token[1:-1])
```

I couldn't find any attribute that records the physical line where the `msgstr` content begins, so it doesn't appear that `sphinx-lint` can obtain that information directly from `polib`.

Before working on a PR, I'd like to confirm the preferred implementation approach. Would you rather:

1. compute the correct mapping while processing the PO file, or
2. use another approach to preserve the exact `msgstr` line numbers?

I'm happy to implement whichever approach best fits the project.

**Comment by BHUVANSH855** [comment]
Update: I've worked out a fix for this (Option 2 from above — scanning the
raw `.po` text rather than trusting `entry.linenum` alone), and it's fully
tested. Posting the approach here before opening a PR, in case a different
direction is preferred.

### Approach

Since `polib` doesn't expose the physical line where `msgstr` content
starts, `po2rst()` now scans forward from `entry.linenum` through the raw
file text to find the actual `msgstr` keyword line, then determines
whether the content starts inline (`msgstr "..."`) or on the next line
(`msgstr ""` followed by wrapped continuation lines):

```python
def _find_msgstr_content_start(raw_lines, entry_linenum):
    scan_start = max(entry_linenum - 1, 0)
    for i in range(scan_start, len(raw_lines)):
        line = raw_lines[i]
        stripped = line.lstrip()
        if stripped.startswith("msgstr "):
            match = _MSGSTR_LINE_RE.match(line)
            first_line_has_content = bool(match and match.group(1))
            return (i + 1) if first_line_has_content else (i + 2)
        if stripped.startswith("msgstr["):
            # Plural forms: fall back to old behaviour for now.
            return entry_linenum
    return entry_linenum
```

`po2rst()` then pads to this computed line instead of `entry.linenum`
directly.

### Result on the reproducer from this issue

```
minimal2.po:7: found an unbalanced inline literal markup. (unbalanced-inline-literals-delimiters)
minimal2.po:16: found an unbalanced inline literal markup. (unbalanced-inline-literals-delimiters)
```

Matches the expected lines exactly.

### Verification done

- Full test suite passes (111 passed, 1 skipped) after updating the two
  existing `test_po2rst.py` cases, which had been asserting the *old,
  incorrect* line numbers as expected output.
- Added a dedicated regression test covering the wrapped-`msgid` case
  from this issue.
- Regression-tested against real translated files (`python-docs-fr` and
  `python-docs-pt-br`'s `library/datetime.po`) — no new false positives.

### Known limitation

Plural forms (`msgid_plural` / `msgstr[N]`) aren't handled precisely yet
and fall back to the old approximate behavior. Happy to scope that as a
separate follow-up rather than block this fix on it, unless you'd rather
it be handled together.

Want me to open this as a PR as-is, or is there a different direction
you'd prefer?

**Comment by BHUVANSH855** [comment]
Gentle bump on this — totally understand things get busy. The fix above
is tested and ready whenever there's time to weigh in on the approach,
or let me know if a different direction would be preferred. No rush,
just didn't want it to fall off the radar.

**Comment by m-aciek** [comment]
Thanks for the investigation. Please open the PR so we can review the implementation. Scanning the raw PO text seems reasonable for this targeted fix. Please make the msgstr match whitespace-aware and keep it bounded to the current entry, and include an end-to-end test for the reported diagnostic lines. It is fine to track plural forms and exact mapping of physically wrapped msgstr fragments separately, but please document those limitations clearly.

As a longer-term improvement, would you consider proposing field-level source locations upstream in polib, such as msgstr_linenum and per-index locations for plural translations? The parser already knows the current physical line while processing these fields, so exposing it would avoid rescanning raw PO text. This should not block the sphinx-lint fix, however; a local fallback remains reasonable until such an API is available.
